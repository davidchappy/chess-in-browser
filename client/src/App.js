import React, { Component } from 'react';
import axios from 'axios';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  constructor(props) {
    super(props);
  
    this.state = {
      white: undefined,
      black: undefined,
      game: undefined
    };
  }

  componentDidMount() {
      axios.post('/api/games', {
        guest1: "Fred",
        guest2: "Ethel"
      })
      .then((response) => {
        console.log(response.data)
        this.setState({
          white: response.data.white,
          black: response.data.black,
          game: response.data.game
        })
      })
      .catch((error) => {
        console.log(error)
      })
  }

  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
