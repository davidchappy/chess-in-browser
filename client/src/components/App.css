.App {
  text-align: center;
}

.App-logo {
  height: 80px;
}

.App-logo.loading {
  animation: App-logo-spin infinite 10s linear;
}

.App-logo.loaded {
  animation: App-logo-pulse 0.5s;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

@keyframes App-logo-pulse {
  0% { transform: scale(1, 1) rotate(0deg); }
  50% { transform: scale(1.2, 1.2) rotate(45deg); }
  100% { transform: scale(1, 1) rotate(0deg); }
}