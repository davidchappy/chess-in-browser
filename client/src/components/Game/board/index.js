import React from 'react';
import './board.css';
import Tile from './tile';

const Board = (props) => {

  // create 2-dim. array with tile rows from game.board data
  const tileNames = Object.keys(props.board);
  let row = [], allRows = [];
  for(let i=0; i<tileNames.length; i++) {
    let tile = tileNames[i];
    if(i === 0 || i % 8 !== 0) {
      row.push(tile);
    } else {
      allRows.push(row);
      row = [];
      row.push(tile);
    }
    if (i === tileNames.length-1) {
      allRows.push(row);
    } 
  }

  const allPieces = props.pieces;
  // map through 2dim array of tiles and return rows of Tiles
  const tiles = allRows.map((rowArray, i) => {
    let rowType = i % 2 === 0 ? 0 : 1;
    let row = rowArray.map((tile, i) => {
      const color = i % 2 === rowType ? 'light' : 'dark';
      let piece = null;
      if(props.board[tile] !== "") {
        for(let pieceNo in allPieces) {
          if(allPieces[pieceNo].position === tile) {
            piece = allPieces[pieceNo];
          }
        }
      }
      const selectable = props.selectableTiles.includes(tile) ? true : false; 
      return (<Tile {...props} tile={tile} key={i} 
                    color={color} piece={piece} 
                    selectable={selectable} />)
    })
    return (
      <div key={i} className="tileRow">
        {row}
      </div>
    )
  });

  return (
    <div className="Board">
      <div className="tiles">
        {tiles}
      </div>
    </div>
  )

}

export default Board;