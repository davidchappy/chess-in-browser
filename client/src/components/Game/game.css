.Game {
  background-color: #D19255;
  height: 100vh;
  width: 100vw;
  text-align: center;
}