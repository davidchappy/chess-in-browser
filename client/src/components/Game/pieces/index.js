import React from 'react';
import Piece from './piece';

const Pieces = (props) => {
  return (
    <div>
      <Piece type="king"/>
    </div>
  )
};

export default Pieces;