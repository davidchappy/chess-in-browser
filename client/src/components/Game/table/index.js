import React from 'react';
import Board from '../board';
import './table.css';

const Table = (props) => {
  return (
    <div className="Table">
      <Board />
    </div>
  )
};

export default Table;