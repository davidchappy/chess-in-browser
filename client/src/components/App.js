import React, { Component } from 'react';
import { connect } from 'react-redux';
import { createGuestGame } from '../actions/chessActions';
import './App.css';
// import { push } from 'react-router-redux';
import { bindActionCreators } from 'redux';
// import { Route, Switch } from 'react-router-dom';
import Game from './Game';
import Welcome from './Welcome';

export class App extends Component {
  constructor(props) {
    super(props);
  
    this.state = {};
  }

  render() {
    let page;

    if(this.props.fetched) {
      page = () => {
        return (
          <Game />
        )
      }
    } else {
      page = () => {
        return (
          <Welcome fetching={this.props.fetching} 
                   createGuestGame={this.props.createGuestGame}/>
        )
      }
    }

    return (
      <div className="App">
        {page()}
      </div>
    );
  }
}

//         <Switch>
//   <Route exact path='/' component={Welcome} />
//   <Route exact path='/game' component={Game} />
// </Switch>

const mapStateToProps = (state, ownProps) => {
  return {
    fetching: state.chess.fetching,
    fetched: state.chess.fetched,
  }
}

const mapDispatchToProps = dispatch => bindActionCreators({
  createGuestGame
}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App)