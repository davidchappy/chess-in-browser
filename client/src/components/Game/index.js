import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import './Game.css';
import Table from './table/';
import Header from '../shared/header';
import { selectPiece, movePiece } from '../../actions/chessActions';

export class Game extends Component {
  constructor(props) {
    super(props);
  
    this.state = {};
  }
  
  render() {
    return (
      <div className="Game">
        <Header {...this.props} headerSize="small" />
        <Table {...this.props} />
      </div>  
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    game: state.chess.game,
    white: state.chess.white,
    black: state.chess.black,
    moving: state.chess.moving,
    selectedPiece: state.chess.selectedPiece,
    selectableTiles: state.chess.selectableTiles,
    fetching: state.chess.fetching,
    fetched: state.chess.fetched
  }
}

const mapDispatchToProps = dispatch => bindActionCreators({
  selectPiece,
  movePiece
}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Game)
