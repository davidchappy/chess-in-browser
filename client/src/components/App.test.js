import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { App } from './App';
import { Game } from './Game/';

describe('rendering App (Snapshot Test)', () => {
  const component = renderer.create(<App />);
  const json = component.toJSON();

  it('renders without crashing', () => {
    expect(json).toMatchSnapshot();
  });

  it('renders Welcome by default', () => {
    expect(json.children[0].props['className']).toBe('Welcome');
  });
});

describe('rendering App (mount test)', () => {
  it('renders Game if server data has been fetched', () => {
    const appWithGameWrapper = shallow(<App fetched={true} />);
    expect(appWithGameWrapper.node.props['children']['type']['displayName']).toBe('Connect(Game)');
  });
});