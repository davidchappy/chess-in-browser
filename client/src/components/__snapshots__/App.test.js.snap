// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering App (Snapshot Test) renders without crashing 1`] = `
<div
  className="App"
>
  <div
    className="Welcome"
  >
    <div
      className="chess-header large"
    >
      <img
        alt="logo"
        className="chess-logo loaded large"
        src="logo.svg"
      />
      <h2>
        Welcome to Chess
      </h2>
    </div>
    <p
      className="Welcome-intro"
    >
      <button
        onClick={[Function]}
      >
        Start Game As Guest
      </button>
    </p>
  </div>
</div>
`;
